### PR TITLE
Refactoring of the  pods env variable to simplify the code

### PR DESCRIPTION
**Pull request by edcdavid**
Extending the PodWrapper struct to completely replace the pods environment variable and simplify the code. Renaming container's StringShort() stringer to String()

### CODE REVIEW

File: pkg/provider/provider.go [hunk 1]
```diff
@@ -51,44 +51,65 @@ const (
 )
 
 type PodWrapper struct {
```
Comment (greyerof) [inline pkg/provider/provider.go:53]: As our custom container wrapper is called simply "Container", cannot we just simply name this one "Pod" ?

**Review by greyerof — APPROVED** [review]
Minor detail for all traces: there's no need to explicitly use put/cut.String() for our custom wrappers if they have a String() method implemented, as `String()` is the method of the stringer interface that will be actually used when golang libs need to format any object using "%s" or "%v". So you could do this:
```
logrus.Infof("Test pod: %s - Test container: %s", put, cut)
```

**Review by sebrandon1 — APPROVED** [review]

File: pkg/provider/provider.go (hunk 1 above)
Comment (edcdavid) [reply pkg/provider/provider.go]: fixed

**Comment by edcdavid** [comment]
> Minor detail for all traces: there's no need to explicitly use put/cut.String() for our custom wrappers if they have a String() method implemented, as `String()` is the method of the stringer interface that will be actually used when golang libs need to format any object using "%s" or "%v". So you could do this:
> 
> ```
> logrus.Infof("Test pod: %s - Test container: %s", put, cut)
> ```

removed all calls to String() in logs, good catch !

**Pull request merged**